### DISCUSSION TITLE
Schalter: Szene für "aus"?

### DISCUSSION THREAD

**Discussion by t1ll** [brutella/hkknx-public#301] [Q&A] [answered]
Möglicherweise stehe ich auf dem Schlauch und habe etwas falsch verstanden, aber soweit ich sehe gibt es aktuell bei Schaltern nur die Möglichkeit eine Szenennummer beim aktivieren zu senden. Was passiert beim deaktivieren des Schalters und wäre es allenfalls eine Idee, dass man in HKKNX auch eine Szenennummer für "AUS" definieren kann?

Danke für jede Hilfe

**Comment by brutella** [comment, marked as answer]
Wenn der Schalter mit einer Szenennummer konfiguriert wird, wird beim Einschalten des Schalters nach 1 Sekunde der Schalter automatisch wieder ausgeschaltet – steht so auch im [Online Handbuch](https://hochgatterer.me/de/hkknx/docs/accessories/switch/). 

Das wurde so umgesetzt, da in KNX eine Szene kein Zustand ist sondern eine Aktion ist.

**Reply by t1ll** [reply]
>  Der Schalter schaltet sich nach dem Senden des Wertes automatisch wird aus.
Aiaiai, hier stehts 🙈 Sorry.

In Homekit ist eine Szene ja "deaktivierbar", deswegen dachte ich, es könnte allenfalls spannend sein, dieses Feature zu replizieren und die Möglichkeit zu bieten, bei AUS auch eine Szene zu senden uns somit das Homekit Verhalten beizubehalten und KNX kompatibel zu machen.

Mit dem Auto-AUS nach einer Sekunde kopiert der Schalten dann auch recht stark das verhalten des Tasters.

**Reply by brutella** [reply]
Ich sehe da ein Problem beim Initialisieren des Tasters. 

- Wie soll der Taster wissen, dass eine Szene aktiv ist? Der Taster wäre somit nur aktiv, wenn er über HomeKit betätigt wurde.
- Was passiert, wenn die Szene von einem KNX Taster "aktiviert" wurde? Es gibt ja kein Rückmelde-Objekt für Szenen, oder etwa doch?

**Reply by t1ll** [reply]
Stimmt, du hast recht, das wäre sehr komplex zu lösen. 

Ich habe bei mir 1-2 solche (physischen) Taster via Logiken gelöst, bei denen Logiken anhand von Statusobjekten entscheiden ob eine Szene als aktiv interpretiert werden soll. Aber ich denke das ist eher ein edge case und daher für hkknx nicht relevant :)
